Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionDeclarativeNetRequestRule.mm

```diff
@@ -334,6 +338,25 @@ - (instancetype)initWithDictionary:(NSDictionary *)ruleDictionary errorString:(N
         }
     }
 
+    if (NSArray<NSString *> *initiatorDomains = _condition[ruleConditionInitiatorDomainsKey]) {
+        if (!isArrayOfDomainsValid(initiatorDomains)) {
+            if (outErrorString)
+                *outErrorString = [NSString stringWithFormat:@"Rule with id %ld is invalid. `initiatorDomains` must be non-empty and cannot contain non-ASCII characters.", (long)_ruleID];
+
+            return nil;
+        }
+    }
+
+    if (NSArray<NSString *> *excludedInitiatorDomains = _condition[ruleConditionExcludedInitiatorDomainsKey]) {
+        if (!isArrayOfDomainsValid(excludedInitiatorDomains)) {
+            if (outErrorString)
+                *outErrorString = [NSString stringWithFormat:@"Rule with id %ld is invalid. `excludedInitiatorDomains` must be non-empty and cannot contain non-ASCII characters.", (long)_ruleID];
```

Comment:
Empty is allowed, per MDN. "If the list is empty or omitted, no domains are excluded."
```suggestion
        if (!isArrayOfExcludedDomainsValid(excludedInitiatorDomains)) {
            if (outErrorString)
                *outErrorString = [NSString stringWithFormat:@"Rule with id %ld is invalid. `excludedInitiatorDomains` cannot contain non-ASCII characters.", (long)_ruleID];
```

Reply:
Whoops, missed the part in the error message here.